app/web3/utils.ts

```diff
@@ -93,3 +93,11 @@ export function computeTotalTxCost(amount: BigNumber): BigNumber {
 
   return amount.add(gasLimit.mul(gasPrice));
 }
+
+export function calculateValueAfterPenalty(eth: string, penaltyRate: string): string {
```

Comment:
I hope you share code of calculations displayed in the Agreement and on Unlock page